--- libsmb2-libsmb2-6.2/lib/krb5-wrapper.c.orig	2024-12-23 04:38:55.000000000 +0100
+++ libsmb2-libsmb2-6.2/lib/krb5-wrapper.c	2025-06-04 21:29:14.953938736 +0200
@@ -57,7 +57,7 @@
 #include <sys/unistd.h>
 #endif
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 #if __APPLE__
 #include <GSS/GSS.h>
 #else
@@ -283,7 +283,7 @@ krb5_negotiate_reply(struct smb2_context
                 return NULL;
         }
 
-        #ifndef __APPLE__ /* gss_set_neg_mechs is not defined on macOS/iOS. */
+        #if !defined(__APPLE__) && 0 /* gss_set_neg_mechs is not defined on macOS/iOS/heimdal. */
         if (smb2->sec != SMB2_SEC_UNDEFINED) {
                 gss_OID_set_desc wantMech;
 
--- libsmb2-4.0.0/configure.ac.orig	2022-01-30 20:20:16.801567540 +0100
+++ libsmb2-4.0.0/configure.ac	2022-01-30 20:26:18.902939202 +0100
@@ -31,7 +31,7 @@ AC_ARG_WITH([libkrb5],
                              NTLMSSP module for authentication instead.])])
 
 AS_IF([test "x$with_libkrb5" != "xno"], [
-    MAYBE_LIBKRB5="-lgssapi_krb5"
+    MAYBE_LIBKRB5="-lgssapi"
     AC_DEFINE([HAVE_LIBKRB5], [1], [Whether we use gssapi_krb5 or not])
     AC_MSG_NOTICE([Build with gssapi_krb5 support])
     dnl  Check for gssapi/gssapi.h
